src/utils/javaUtils.ts

```diff
@@ -110,14 +107,75 @@ export namespace javaUtils {
         return node.root.client.updateApplicationSettings(appSettings);
     }
 
-    export async function getJavaFileExtensions(siteConfig: WebSiteManagementModels.SiteConfigResource | undefined): Promise<string | string[] | undefined> {
-        if (siteConfig && isJavaRuntime(siteConfig.linuxFxVersion)) {
-            return getArtifactTypeByJavaRuntime(siteConfig.linuxFxVersion);
+    export async function getJavaFileExtensions(siteConfig: WebSiteManagementModels.SiteConfigResource | undefined): Promise<string[] | undefined> {
+        if (isJavaSERuntime(siteConfig)) {
+            return ['jar'];
+        } else if (isJavaWebContainerRuntime(siteConfig)) {
+            return ['war'];
         } else if (await isJavaProject()) {
             return getJavaArtifactExtensions();
         }
-
         return;
     }
 
+    function getMavenModuleFromPom(pomFile: string): { pom: string, artifactId: string, packaging: string } | undefined {
+        const pomContent = fse.readFileSync(pomFile, 'utf8');
+        try {
+            const pom = parser.parse(pomContent) as { project: { artifactId: string; packaging: string; }; };
+            if (pom.project && pom.project.artifactId) {
+                return {
+                    pom: pomFile,
+                    artifactId: pom.project.artifactId,
+                    packaging: pom.project.packaging || 'jar'
+                } as { pom: string; artifactId: string; packaging: string; };
+            }
+        } catch (e) {
+            return undefined;
+        }
+        return undefined;
+    }
+
+    export function getMavenModule(module: string): { pom: string, artifactId: string, packaging: string } | undefined {
+        if (fse.existsSync(module)) {
+            if (fse.lstatSync(module).isDirectory() && fse.existsSync(path.join(module, 'pom.xml'))) {
+                return getMavenModuleFromPom(path.join(module, 'pom.xml'));
+            } else if (fse.lstatSync(module).isFile() && path.extname(module) === '.xml') {
+                return getMavenModuleFromPom(module);
+            }
+        }
+        return undefined;
+    }
+
+    export async function getLocalMavenWrapper(modulePath: string): Promise<string | undefined> {
+        const mvnw: string = constants.isWindows ? "mvnw.cmd" : "mvnw";
+        // walk up parent folders
+        let current: string = modulePath;
+        while (path.basename(current)) {
+            const potentialMvnwPath: string = path.join(current, mvnw);
+            if (await fse.pathExists(potentialMvnwPath)) {
+                return potentialMvnwPath;
+            }
+            current = path.dirname(current);
+        }
+        return undefined;
+    }
+
+    export async function isExecutableJarFile(fsPath: string): Promise<boolean> {
```

Comment:
Please avoid using a utils file when possible. We really only want to put code in the utils folder if it's shared by several other files, but I think a decent amount of this code is only used in one place.

Reply:
It's OK for me to move this method somewhere else. 

IMO:
we should judge if or not a method/function should be put in a utils file/class based on if the method/function itself is a util or not.
**in other words, a util method/function should be put into utils file/class regardless of it's shared or not for current moment.** in this way, future developers can easily find it when they need.

Reply:
fixed to your pattern